### ISSUE TITLE
ENI Trunking is not working

### ISSUE THREAD

**Issue by igor-pinchuk** [aws/amazon-ecs-agent#2760]
### Summary
AWSVPC Trunking not working on old ECS clusters.

### Description
I have enabled AWSVPC Trunking globally in AWS account, rotated ECS instances several times but still getting ENI resource limit errors, my ECS cluster still supports only 3 ENIs per m5.large instead of promised 10 ENIs.

aws ecs list-attributes --target-type container-instance --attribute-name ecs.awsvpc-trunk-id  --cluster my-cluster-name --region us-east-2
{
    "attributes": []
}

### Environment Details
I am using lates AWS ECS optimized AMI with the latest ECS agent on it.
Agent version 1.48.1
Docker version 19.03.13-ce
AWS region: us-east-2

**Comment by igor-pinchuk** [comment]
It seems like this is the same issue
https://github.com/aws/amazon-ecs-agent/issues/2448

**Comment by jait** [comment]
Experiencing the same issue.
I have enabled the ENI trunking setting for the account and "aws ecs list-account-settings --effective-settings" lists the setting as enabled:
  {
            "name": "awsvpcTrunking",
            "value": "enabled",
            "principalArn": "arn:aws:iam::XXX:role/YYY"
  }

but none of the newly-created m5a.large instances support more than 2 ENIs.

**Comment by fenxiong** [comment]
Hi,
Note that to use ENI trunking, the awsvpcTrunking account setting needs to be enabled specifically for the [container instance role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html) being used for the EC2 instance. Can you try running the following command *on the container instance*  and see if the setting has been enabled:
```
aws ecs list-account-settings --effective-settings --region <region>
```
If it's disabled, then you can enable it with
```
aws ecs put-account-setting --region <region> \
      --name awsvpcTrunking \
      --value enabled
```
and then subsequent container instance launched with the same role should have ENI trunking enabled.

**Comment by igor-pinchuk** [comment]
Thanks @jait and @fenxiong 

My issue was that I thought this setting (awsvpcTrunking) was a global one across all AWS regions and I applied it in us-east-1 (which is my default region) and in us-east-2 where my ECS cluster lives it was still not enabled. I realized it just now that ECS Account Settings are applied per region, not globally, despite they are set for global IAM resources like users and roles.

And yes, awsvpcTrunking needs to be enabled for specific IAM role or "Set for root" scope using root AWS account to override default values.

So my problem is fixed!
